### PR TITLE
Make Zowe Explorer independent of Zowe CLI installation

### PR DESCRIPTION
**Pull request by JillieBeanSim**
Signed-off-by: Billie Simmons <49491949+JillieBeanSim@users.noreply.github.com>

## Proposed changes

This PR will 
1. check for and create if needed the .zowe/settings/imperative.json file making Zowe Explorer independent of the Zowe CLI installation.  
2. fixes the watcher of the Secure Credentials Enabled setting, to update file when checked or unchecked.  
3. will read and parse the file if existing and changing the CredentialManager value then write back. This is done just incase imperative, zowe cli, or an extender/plugin decide to introduce more values to this file.

## Release Notes

Milestone: 2.4.1

Changelog:
- Added independence from Zowe CLI installation by creating the `~/.zowe/settings/imperative.json` file during activation if it doesn't already exist. This file is for Zowe Explorer to know the Security Credential Manager used for secure profile information.
- Fixed the `Secure Credentials Enabled` setting to update the `~/.zowe/settings/imperative.json` file upon change of the setting without overwriting other data in the file.

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Further comments

https://user-images.githubusercontent.com/49491949/198721982-c5ea7a94-b192-41da-b173-c7b4c0c94da4.mov

https://user-images.githubusercontent.com/49491949/198725505-f1504304-2a08-4b19-bfe2-2859860224e8.mov

### CODE REVIEW

**Comment by JillieBeanSim** [comment]
The code smell is about the comment with `TODO` in it, which I think can stay until the issue mentioned is solved and we consume and use the changes.

File: packages/zowe-explorer/__tests__/__unit__/extension.unit.test.ts [hunk 1]
```diff
@@ -464,7 +470,7 @@ describe("Extension Unit Tests", () => {
 
         expect(globals.ISTHEIA).toEqual(true);
         // tslint:disable-next-line: no-magic-numbers
-        expect(globalMocks.mockMkdirSync.mock.calls.length).toBe(4);
+        expect(globalMocks.mockMkdirSync.mock.calls.length).toBe(6);
```
Comment (zFernand0) [inline packages/zowe-explorer/__tests__/__unit__/extension.unit.test.ts:473]: we should probably remove tslint in favor of eslint.
and for these kind of tests, we could have the eslint configuration for tests exclude the magic-number rules 😋 
Example: https://github.com/zowe/zowe-cli/blob/master/.eslintrc.js#L14-L36

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

File: packages/zowe-explorer/CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,16 @@
 
 All notable changes to the "vscode-extension-for-zowe" extension will be documented in this file.
 
+## TBD Release
+
+### New features and enhancements
+
+- Added independence from Zowe CLI installation by creating the `~/.zowe/settings/imperative.json` file during activation if it doesn't already exist. This file is for Zowe Explorer to know the Security Credential Manager used for secure profile information. [#1850](https://github.com/zowe/vscode-extension-for-zowe/issues/1850)
+
+### Bug fixes
```
Comment (zFernand0) [inline packages/zowe-explorer/CHANGELOG.md:11]: should we move this entry to the `Bug fixes` section ?

File: packages/zowe-explorer/__tests__/__unit__/extension.unit.test.ts (hunk 1 above)
Comment (JillieBeanSim) [reply packages/zowe-explorer/__tests__/__unit__/extension.unit.test.ts:473]: it doesn't look like Zowe Explorer has changed over to eslint, only ZE API & FTP

Comment (zFernand0) [reply packages/zowe-explorer/__tests__/__unit__/extension.unit.test.ts:473]: 🤦 you are right
the first step is to migrate to eslint 😋

**Review by t1m0thyj — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]

**Pull request merged**